albatross/evaluation/cross_validation_utils.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_CROSS_VALIDATION_UTILS_H
+#define ALBATROSS_CROSS_VALIDATION_UTILS_H
+
+namespace albatross {
+
+template <typename ModelType, typename FeatureType>
+inline auto
+get_predictions(const ModelType &model,
+                const std::vector<RegressionFold<FeatureType>> &folds) {
+
+  using FitType = typename fit_type<ModelType, FeatureType>::type;
+  std::vector<Prediction<ModelType, FeatureType, FitType>> predictions;
+  for (const auto &fold : folds) {
+    predictions.emplace_back(model.get_fit_model(fold.train_dataset)
+                                 .get_prediction(fold.test_dataset.features));
+  }
+
+  return predictions;
+}
+
+template <typename FeatureType>
+inline std::size_t
+max_index(const std::vector<RegressionFold<FeatureType>> &folds) {
```

Comment:
This function and it's use seems a bit hacky. The function finds the highest index in the folds, but contrary to the name, adds 1 to return the number of original features. There's also an assumption that the highest index appears in the fold's test sets, which is true is using a GrouperFunction, but not true of folds themselves.
It's like there should be two types of folds, a generic one and a strict one where the train sets are disjoint, union to a full dataset, ect.

Reply:
yeah I guess I should probably do a full check to make sure the folds completely cover the dataset.